libraries/hotstuff/include/eosio/hotstuff/qc_chain.hpp

```diff
@@ -159,10 +160,12 @@ namespace eosio::hotstuff {
 
       std::vector<hs_proposal_message> get_qc_chain(const fc::sha256& proposal_id); //get 3-phase proposal justification
 
-      void send_hs_proposal_msg(const hs_proposal_message& msg); //send vote msg
-      void send_hs_vote_msg(const hs_vote_message& msg); //send proposal msg
-      void send_hs_new_view_msg(const hs_new_view_message& msg); //send new view msg
-      void send_hs_new_block_msg(const hs_new_block_message& msg); //send new block msg
+      void send_hs_proposal_msg(const hs_proposal_message& msg, bool propagation = false); //send vote msg
+      void send_hs_vote_msg(const hs_vote_message& msg, bool propagation = false); //send proposal msg
+      void send_hs_new_view_msg(const hs_new_view_message& msg, bool propagation = false); //send new view msg
+      void send_hs_new_block_msg(const hs_new_block_message& msg, bool propagation = false); //send new block msg
```

Comment:
I would prefer to be explicit and not have the default `= false`.
Also a couple of the comments are wrong and they are all completely useless as they do not tell you anything more than the method name.